tifs-a-tikv-based-partition-tolerant-strictly-consistent-file-system.md

```diff
@@ -0,0 +1,326 @@
+|        |                                          |                                              |
+|        |                                          |                                              |
+|        |                                          |                                              |
+|        |                                          |                                              |
+|        |                                          |                                              |
+|        v                                          v                                              v
++--------------------------------------------------------------------------------------------------+
+|        |                                          |                                              |
+|    3   |              inode number                |                  file handler                |
+|        |                                          |                                              |
++--------+------------------------------------------+----------------------------------------------+
+```
+
+The **file index** key includes the inode number of its parent directory in big-endian ordering and the file name in UTF-8 encoding:
+
+```
++ 1 byte +<---------------- 8 bytes --------------->+<------------------ 8 bytes ----------------->+
+|        |                                          |                                              |
+|        |                                          |                                              |
+|        |                                          |                                              |
+|        |                                          |                                              |
+|        |                                          |                                              |
+|        |                                          |                                              |
+|        v                                          v                                              v
++--------------------------------------------------------------------------------------------------+
+|        |                                          |                                              |
+|    4   |     inode number of parent directory     |         file name in utf-8 encoding          |
+|        |                                          |                                              |
++--------+------------------------------------------+----------------------------------------------+
+```
+
+
+### Consistency
+
+TiKV supports both optimistic and pessimistic transactions. However, since the Rust client only supports pessimistic transactions experimentally, and pessimistic transactions may reduce performance when transactions don't conflict, **we implement TiFS by optimistic  transactions only**.
```

Comment:
```suggestion
TiKV supports both optimistic and pessimistic transactions. However, since the Rust client only supports pessimistic transactions experimentally, and pessimistic transactions may reduce performance when transactions don't conflict, **we implement TiFS by optimistic transactions only**.
```